src/frontends/tensorflow/src/helper_ops/block_lstm.hpp

```diff
@@ -0,0 +1,158 @@
+        // 6) h A Tensor. Has the same type as x.
+
+        // extract time_len, batch_size
+        auto time_len = ov::Dimension::dynamic();
+        auto batch_size = ov::Dimension::dynamic();
+        auto x_shape = get_input_partial_shape(1);
+        auto x_type = get_input_element_type(1);
+        auto x_rank = x_shape.rank();
+        if (x_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                x_rank.get_length() == 3,
+                "Internal error in OpenVINO TensorFlow Frontend: input data for BlockLSTM must be of rank equal to 3.");
+            time_len = x_shape[0].is_static() ? x_shape[0].get_length() : ov::Dimension::dynamic();
+            batch_size = x_shape[1].is_static() ? x_shape[1].get_length() : ov::Dimension::dynamic();
+            m_input_size = x_shape[2].is_static() ? x_shape[2].get_length() : ov::Dimension::dynamic();
+        }
+        m_batch_size = batch_size;
+
+        // extract hidden_size
+        m_hidden_size = ov::Dimension::dynamic();
+        auto w_shape = get_input_partial_shape(4);
+        auto w_rank = w_shape.rank();
+        auto b_shape = get_input_partial_shape(8);
+        auto b_rank = b_shape.rank();
+        if (w_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                w_rank.get_length() == 2,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for BlockLSTM must be of rank equal to 2.");
+            m_hidden_size = w_shape[1].is_static() ? w_shape[1].get_length() / 4 : ov::Dimension::dynamic();
+        } else if (b_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                b_rank.get_length() == 1,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for BlockLSTM must be of rank equal to 2.");
+            m_hidden_size = b_shape[0].is_static() ? b_shape[0].get_length() / 4 : ov::Dimension::dynamic();
+        }
+
+        // set shapes of dynamic rank for inputs except 1 and 6 which OpenVINO LSTMSequence supports now
+        // other outputs are out of scope
+        set_output_type(0, x_type, ov::PartialShape::dynamic());
+        set_output_type(1, x_type, ov::PartialShape{time_len, batch_size, m_hidden_size});
+        set_output_type(2, x_type, ov::PartialShape::dynamic());
+        set_output_type(3, x_type, ov::PartialShape::dynamic());
+        set_output_type(4, x_type, ov::PartialShape::dynamic());
+        set_output_type(5, x_type, ov::PartialShape::dynamic());
+        set_output_type(6, x_type, ov::PartialShape{time_len, batch_size, m_hidden_size});
+    }
+
+    float get_forget_bias() const {
+        return m_forget_bias;
+    }
+
+    float get_cell_clip() const {
+        return m_cell_clip;
+    }
+
+    bool get_use_peephole() const {
+        return m_use_peephole;
+    }
+
+    ov::Dimension get_hidden_size() const {
+        return m_hidden_size;
+    }
+
+    ov::Dimension get_attr_input_size() const {
+        return m_input_size;
```

Comment:
Would be nice to align function name with variable name. Same for `m_batch_size`.

Reply:
I removed them. No longer needed. They become a part of sub-graph to preserve re-shapebility